pkg/server/server.go

```diff
@@ -474,47 +485,68 @@ func (s *ProxyServer) Connect(stream agent.AgentService_ConnectServer) error {
 
 // route the packet back to the correct client
 func (s *ProxyServer) serveRecvBackend(stream agent.AgentService_ConnectServer, agentID string, recvCh <-chan *client.Packet) {
+	defer func() {
+		// Close all connected frontends when the agent connection is closed
+		// TODO(#126): Frontends in PendingDial state that have not been added to the
+		//             list of frontends should also be closed.
+		frontends, _ := s.getFrontends(agentID)
```

Comment:
This might be problematic.

In HA setup, the same agent might end up with multiple streams to the proxy server when it tries to establish streams to all server instances. The agent will close the redundant streams. However, this cleanup function will close all frontends that connect via the agent, even if the supporting `stream` is not closed.

I think we should track frontends based on streams, and cleanup frontends based on streams.